packages/core/contracts/financial-templates/optimistic-distributor/OptimisticDistributor.sol

```diff
@@ -0,0 +1,425 @@
+        bytes32 merkleRoot,
+        string ipfsHash
+    );
+    event ProposalRejected(uint256 indexed rewardIndex, uint256 indexed proposalIndex);
+    event ProposalDeleted(uint256 indexed rewardIndex, uint256 indexed proposalIndex);
+    event MerkleDistributorSet(address indexed merkleDistributor);
+
+    /**
+     * @notice Constructor.
+     * @param _bondToken ERC20 token that the bond is paid in.
+     * @param _finder Finder to look up UMA contract addresses.
+     * @param _timer Contract that stores the current time in a testing environment.
+     */
+    constructor(
+        FinderInterface _finder,
+        IERC20 _bondToken,
+        address _timer
+    ) Testable(_timer) {
+        finder = _finder;
+        require(_getCollateralWhitelist().isOnWhitelist(address(_bondToken)), "Bond token not supported");
+        bondToken = _bondToken;
+        syncUmaEcosystemParams();
+    }
+
+    /********************************************
+     *            FUNDING FUNCTIONS             *
+     ********************************************/
+
+    /**
+     * @notice Allows any caller to create a Reward struct and deposit tokens that are linked to these rewards.
+     * @dev The caller must approve this contract to transfer `maximumRewardAmount` amount of `rewardToken`.
+     * @param rewardToken ERC20 token that the rewards will be paid in.
+     * @param maximumRewardAmount Maximum reward amount that the sponsor is posting for distribution.
+     * @param earliestProposalTimestamp Starting timestamp when proposals for distribution can be made.
+     * @param priceIdentifier Identifier that should be passed to the Optimistic Oracle on proposed distribution.
+     * @param customAncillaryData Custom ancillary data that should be sent to the Optimistic Oracle on proposed
+     * distribution.
+     * @param optimisticOracleProposerBond Amount of bondToken that should be posted in addition to final fee
+     * to the Optimistic Oracle on proposed distribution.
+     * @param optimisticOracleLivenessTime Liveness period in seconds during which proposed distribution can be
+     * disputed through Optimistic Oracle.
+     */
+    function createReward(
+        uint256 maximumRewardAmount,
+        uint256 earliestProposalTimestamp,
+        uint256 optimisticOracleProposerBond,
+        uint256 optimisticOracleLivenessTime,
+        bytes32 priceIdentifier,
+        IERC20 rewardToken,
+        bytes calldata customAncillaryData
+    ) external nonReentrant() {
+        require(address(merkleDistributor) != address(0), "Missing MerkleDistributor");
+        require(_getIdentifierWhitelist().isIdentifierSupported(priceIdentifier), "Identifier not registered");
+        require(_ancillaryDataWithinLimits(customAncillaryData), "Ancillary data too long");
+        require(optimisticOracleLivenessTime >= MINIMUM_LIVENESS, "OO liveness too small");
+        require(optimisticOracleLivenessTime < MAXIMUM_LIVENESS, "OO liveness too large");
```

Comment:
nit: this should be `<=` right? When I read `MAXIMUM_LIVENESS` I read that the value is allowable, but nothing greater

Reply:
No, OO does not support that: https://github.com/UMAprotocol/protocol/blob/d472e177c3485d1fbb149e10a61165d41199347f/packages/core/contracts/oracle/implementation/OptimisticOracle.sol#L622

Reply:
but not sure how to rename MAXIMUM_LIVENESS properly to reflect this?